R/checks.R

```diff
@@ -0,0 +1,46 @@
+#' Validate data input
+#'
+#' @description
+#' `check_reports_valid()` checks that the input data frame is a data frame,
+#' and that it has the right column names and types. In particular, it checks
+#' that the date column is in date format and does not contain NA's, and that
+#' the other columns are numerics.
+#'
+#' @param reports A data frame with either:
+#' * two columns: date and confirm, if to be
+#'  used by estimate_infection() or estimate_truncation, or
+#' * three columns: date, primary, and secondary, if to be used by
+#'  estimate_secondary().
+#' @param for_estimate_secondary Logical; whether the data is being passed by
+#' estimate_secondary() or the other estimate_*() functions.
+#' @importFrom checkmate assert_data_frame assert_date assert_names
+#' assert_numeric
+#' @return Called for its side effects.
+#' @author James M. Azam
+#' @keywords internal
+check_reports_valid <- function(reports, for_estimate_secondary) {
```

Comment:
Following the argument at https://design.tidyverse.org/boolean-strategies.html (as pointed out by @seabbs) I think it might be better to have this as an enum, something along the lines of `for = c("estimate_infections", "estimate_secondary")`

Reply:
I've made this change here b4c6ecb46882fdc54b40a0be7a0ea2feb32b0636.